activesupport/lib/active_support/message_verifier.rb

```diff
@@ -106,18 +118,36 @@ def verify(signed_message)
     #
     #   verifier = ActiveSupport::MessageVerifier.new 's3Krit'
     #   verifier.generate 'a private message' # => "BAhJIhRwcml2YXRlLW1lc3NhZ2UGOgZFVA==--e2d724331ebdee96a10fb99b089508d1c72bd772"
-    def generate(value)
-      data = encode(@serializer.dump(value))
-      "#{data}--#{generate_digest(data)}"
+
+    def generate(options)
+      @claims = Claims.new(options)
+      data = encode(encoded_header, encoded_claims).join('.')
```

Comment:
actually ```encoded_header``` and ```encoded_claims``` are just serialized versions of header and claims. I will change the name of these functions. ```encode``` is for Base64 encoding :smile: